core/Enclave/openssl_operation.cpp

```diff
@@ -342,7 +352,7 @@ sgx_status_t sm4_cbc_decrypt(uint8_t *key,
     int temp_len = 0;
     EVP_CIPHER_CTX *pctx = NULL;
 
-    int pad = (ciphertext_len % 16 == 0) ? 0 : 1;
+    int pad = (ciphertext_len % 16 == 0) ? SM4_NO_PAD : SM4_PAD;
```

Comment:
It seems that the ciphertext_len will always satisfy `ciphertext_len % 16 == 0` because:

- If `len(text_before_encryption) % 16 == 0`, the `ciphertext_len % 16 == 0`
- If `len(text_before_encryption) % 16 != 0`, the encryption will add padding to ensure `len(text_before_encryption) % 16 == 0`

Reply:
New code is upload and I try to use EVP api only to solve the issue
EVP_DecryptFinal_ex should unpad auto and it seems work well in unittest test